src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -354,7 +354,15 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
                             throw new RuntimeException(sprintf('Cannot create an instance of %s from serialized data because the variadic parameter %s can only accept an array.', $class, $constructorParameter->name));
                         }
 
-                        $params = array_merge($params, $data[$paramName]);
+                        $variadicParameters = array_map(
+                            function ($parameterData) use ($reflectionClass, $constructorParameter, $paramName, $context, $format) {
```

Comment:
Can you use a foreach loop instead of `array_map`? It's a bit faster, and it can matter here because it's on the hot path.